tidy3d/web/api/webapi.py

```diff
@@ -134,6 +139,18 @@ def run(
     :meth:`tidy3d.web.api.container.Batch.monitor`
         Monitor progress of each of the running tasks.
     """
+    if use_local_cache and Path(path).is_file():
+        local_sim = SimulationData.from_file(path)
+        if local_sim.simulation.sha256 == simulation.sha256:
```

Comment:
Our `sha256` is just the hdf5 bytes stream, so it *could* be possible to get those without loading the SimulationData, getting the simulation, and calling `sha256` on it. The `simulation` group in the `SimulationData` hdf5 should just be the same bytes as a `Simulation.to_hdf5()` would produce?

Reply:
That's true, but it would require a bunch more logic for getting the hdf5 stream and calculating the hash directly on that, basically reimplementing what's already in the base model? If we instead add the calculated hash to the simulation data directly (e.g. in `download()`), that might not be necessary?

Reply:
Yes, not sure what you mean in `download()` though?

Also it would still be nice to load the hash without doing `SimulationData.from_file`, but yeah it should be possible to read directly from the hdf5.

Reply:
The idea was to "intercept" the simulation data when it is being downloaded, hashing the simulation, and putting the hash in `.attrs` of the `SimulationData` so that it's available without needing to recalculate the hash. But I don't think that's feasible because there is no client-side assembly of the simulation data happening at all, so if we went that route, then that would need to be implemented on the backend. I can do the bytes thing though.

Reply:
I see, but maybe it makes more sense in `load` rather than in `download`, because `download` should purely download the file without actually loading the simulation data.

Reply:
Sorry, that's kind of like what you are saying in the second part of your comment. :) But yeah it could be done in the `load` function... but I don't think I like this route because I don't like the idea of the user-side and server-side files differing. If we want to do it we should probably just do it upon simulation init, or at least upload.

Reply:
This is proving to be a bit more tricky than I thought. The problem is that for single simulations, I can hash the byte stream just fine since it just means reading the file. However, for simulation data files, the file is split into the groups `data` and `simulation`, the latter being whose byte stream I need to hash. However, I haven't found a way to load a single hdf5 group as a byte stream from disk. I don't think this is even possible because hdf5 does not store groups contiguously (only datasets). So the only way I can see right now to hash the simulation correctly is to extract the bytestream as it is written _to_ disk, which does involve reconstructing the objects. Any ideas? Otherwise I'd perhaps just leave it like this for now, it's still way faster than going through the web pipeline.